src/models/invites-ignorer.ts

```diff
@@ -315,16 +351,18 @@ export class IgnoredInvites {
     private async withIgnoreInvitesPolicies(cb: (ignoreInvitesPolicies: {[key: string]: any}) => void) {
         const { policies, ignoreInvitesPolicies } = this.getPoliciesAndIgnoreInvitesPolicies();
         cb(ignoreInvitesPolicies);
+        await this.withIgnoreInvitesPoliciesPromise;
         policies[IGNORE_INVITES_ACCOUNT_EVENT_KEY.name] = ignoreInvitesPolicies;
-        await this.client.setAccountData(POLICIES_ACCOUNT_EVENT_TYPE.name, policies);
+        this.withIgnoreInvitesPoliciesPromise = this.client.setAccountData(POLICIES_ACCOUNT_EVENT_TYPE.name, policies);
+        return this.withIgnoreInvitesPoliciesPromise;
     }
 
     /**
      * As `getIgnoreInvitesPolicies` but also return the `POLICIES_ACCOUNT_EVENT_TYPE`
-     * object.
+     * object. Content is NOT validated.
      */
     private getPoliciesAndIgnoreInvitesPolicies():
-        {policies: {[key: string]: any}, ignoreInvitesPolicies: {[key: string]: any}} {
+        {policies: any, ignoreInvitesPolicies: any} {
```

Comment:
these also need types

Reply:
Same response, sadly. If there is a better way to mark untrusted-deserialized-json than `any`, I'll be happy to use it!